hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/ECAdmin.java

```diff
@@ -642,6 +642,10 @@ public int run(Configuration conf, List<String> args) throws IOException {
           throw e;
         }
       } else {
+        if (args.size() > 0) {
+          System.err.println(getName() + ": Too many arguments");
```

Comment:
The error message here is unfriendly, let me think about how to handle it.

Reply:
`System.err.println(getName() + ": Input invalid arguments.\nUsage: " + getLongUsage());` 
How about it?

Reply:
1. It looks good!
However, if multiple EC policies are enabled in the cluster, use `bin/hdfs ec -verifyClusterSetup` command is incorrect.

2. The reasons are as follows:
`DistributedFileSystem#getECTopologyResultForPolicies` can only return one result (ECTopologyVerifierResult).
```java
  /**
   * Verifies if the given policies are supported in the given cluster setup.
   * If not policy is specified checks for all enabled policies.
   * @param policyNames name of policies.
   * @return the result if the given policies are supported in the cluster setup
   * @throws IOException
   */
  public ECTopologyVerifierResult getECTopologyResultForPolicies(
      final String... policyNames) throws IOException {
    return dfs.getECTopologyResultForPolicies(policyNames);
  }
```

3. Modify to require specifying the `-policy` param? Or modify the return value of `DistributedFileSystem#getECTopologyResultForPolicies`?

Reply:
It works as designed, if you don't specify any policy the result is if all the policies enabled can work or not & the minimum number of datanodes required for each of them.

It is validating cluster setup for the enabled policies, not per policy setup.

if you specify multiple policies, it provides the aggregate result, like if all those policies can work on the cluster or not & if not a combined message like how man DN are required.
like
```
9 DataNodes are required for the erasure coding policies: RS-6-3-1024k, XOR-2-1-1024k. The number of DataNodes is only 3. 
```

Like here, 9 is required number because of 6-3.

If we intend to improve the behaviour we can think if there is a need

Reply:
@ayushtkn Thx for your Q&A, this is the design of the API itself.
However, the combined message cannot clearly explain the reasons why per EC policy is not supported.

Reply:
If you pass multiple policies, that means you want a combined result, like if all these are supported or not in the "cluster", if you want one policy, pass one policy only.

The whole design is to verify things at cluster level, not at policy level. To highlight cluster level setup issues, like all the enabled policies aren't supported & things like that, it was created for cluster admin level usage.

you can add an additional option which tells the result per policy if the additional option is provided, in that case post getting the result, it can loop over the policies and get individual result, changing proto & all is like last things to do, but I don't think it is required as of now.

Reply:
Thx, I understand. It is necessary to modify the protocol in order to support policy level validation.

Reply:
Agree with @ayushtkn , the current works as designed, if don't specify any policy the result is if validate all the enabled policies.
maybe the current ticket only fix verifyClusterSetup an invalid param output ,thanks~

Reply:
Thx, @haiyang1987, No problem! 
Changing the error message:
```java
System.err.println(getName() + ": Input invalid arguments.\nUsage: " + getLongUsage());
```
How about it? cc @ayushtkn

Reply:
we can change but it won't be consistent with others
https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/ECAdmin.java#L115-L117
https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/ECAdmin.java#L115-L117
https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/ECAdmin.java#L115-L117

Reply:
@ayushtkn, That won't change for now. Thanks your guidance!